### ISSUE TITLE
Also make it possible to change background colors

### ISSUE THREAD

**Issue by Weber-Frank** [Ekopalypse/EnhanceAnyLexer#11]
Is there a way to extend EnhanceAnyLexer so that background colors can also be changed?
Personally, it would be enough for me if only the background colors could be changed, so it might be possible to program a switch like use RGB instead of BGR for using background colors instead of foreground colors.
Of course, a complete solution for foreground and background colors would be better, but it would definitely require a change to the ini file, so there could be incompatibilities with existing configurations for other users.
I would be happy if something like this could be made possible.

Thank you for the work you have put into the project so far.
Best regards

**Comment by Ekopalypse** [comment]
The background colors of the individual matches? Like we say Jein (Yes and no).

It is possible to use indicators such as [INDIC_STRAIGHTBOX](https://www.scintilla.org/ScintillaDoc.html#SCI_INDICSETSTYLE) or INDIC_ROUNDBOX,
but these can only be used with permanently assigned colors and not
be used at runtime like INDIC_TEXTFORE with [SC_INDICFLAG_VALUEFORE ](https://www.scintilla.org/ScintillaDoc.html#SCI_INDICSETFLAGS).

Currently 44 different indicators can be used with Scintilla, some of which are used by Scintilla itself, from 32 upwards, 0-7 are assigned to the lexers and Npp and other plugins can/should use the rest.
In other words, you don't know beforehand how many can actually be used without conflict. If it's just one or two fixed indicators, then there's a good chance you can get those assigned, but 10 or 20 ... probably not.

**Comment by Weber-Frank** [comment]
Thanks for your fast answer.
Jein 😂 I understand that. I'm German.

I thought it would be easier to assign the background colors.
Maybe it is possible to use INDIC_STRAIGHTBOX together with the colors "Mark Style 1" to "Mark Style 5" defined in stylers.xml?
This should ensure that the color is available.
An entry in the ini file could then look like this:
mark_style_1 = test
mark_style_2 = peace

This probably wouldn't cause any conflicts with existing ini files.
Bt I don't know whether it's easy to get the Mark Style colors from the configuration into the plugin or whether the stylers.xml has to be read and evaluated from the plugin.

Best regards

**Comment by Ekopalypse** [comment]
Sorry for replying so late, but a flu-like infection caught me at the weekend. Of course ... at the weekend ... grml :-)

> I'm German.

Haha ... me too.


Using the "style token" function with EnhanceAnyLexer has an annoying effect.
Once defined, the document types handled by the plugin can no longer use the "style token" Npp function.

![image](https://github.com/Ekopalypse/EnhanceAnyLexer/assets/47723516/45c5e181-2ea2-43b2-a2ec-d5c537e23457)

**Comment by Weber-Frank** [comment]
Firstly, it is not good that you are sick and I hope you get well soon.
I don't know what -25 and -25 mean in your example. I suspect this is your trial version to test this functionality.

You seem to misunderstand me. IIRC, the problem was assigning possible colors, so my suggestion was to take the bgColor from Mark Style 1 to Mark Style 5 defined in stylers.xml and use those colors in the plugin.
I didn't suggest calling the style token function.
Hopefully using these colors to color the background shouldn't be a problem with the plugin.

Get well soon and kind regards

**Comment by Ekopalypse** [comment]
> Firstly, it is not good that you are sick and I hope you get well soon

Thanks, it's getting better by the day.

Oh no, I didn't use the style token function itself, I reused the indicator that is basically responsible for doing this. Maybe a short gif will illustrate this a little better.


![enhance_any_lexer_backgrund_styling](https://github.com/Ekopalypse/EnhanceAnyLexer/assets/47723516/86d29073-f12d-4df6-8b89-acdc8c427026)

As you can see, it is automatically colored, but you can no longer use the style token functions for the 1st and 2nd style. Btw 25, 24 (and 23, 22, 21) are the indicator IDs that Npp defines. I chose to use -25, -24 to easily distinguish them from the normal foreground color regexes.
I'm not sure if I should go this route. Ultimately it's up to the users to decide what's more helpful in their cases, but ... maybe I should rethink the whole idea with the indicators and go back and re-investigate the styles that caused some problems in an older version.

**Comment by Weber-Frank** [comment]
Thank you for your excellent explanation and illustration and sorry for the late reply, I was out of town.
I didn't think it was that complicated. I thought N++ makes its style dependent on the language and then the plugin examines the text and if an applicable regular expression is found it applies the style to the appropriate text in addition to the already formatted text.
By the way, I could live with the restriction you mentioned.
Thank you for your efforts, maybe you will find a good solution. I hope I'm not the only one who would like a feature like this.
Best regards.

**Comment by Ekopalypse** [comment]
@Weber-Frank - I will close this one and continue to document the status in #12.

**Comment by Weber-Frank** [comment]
👍
